cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/file/python/PythonFileConceptPass.kt

```diff
@@ -0,0 +1,368 @@
+                }
+                "os.chmod" -> {
+                    val fileName =
+                        callExpression.argumentValueByNameOrPosition<String>(
+                            name = "path",
+                            position = 0,
+                            evaluator = PythonValueEvaluator(),
+                        ) as? String
+                    if (fileName == null) {
+                        Util.errorWithFileLocation(
+                            callExpression,
+                            log,
+                            "Failed to parse the `path` argument. Ignoring the entire `os.chmod` call.",
+                        )
+                        return
+                    }
+
+                    val file = getOrCreateFile(fileName, callExpression)
+
+                    val mode =
+                        callExpression.argumentValueByNameOrPosition<Long>(
+                            name = "mode",
+                            position = 1,
+                            evaluator = PythonValueEvaluator(),
+                        )
+                    if (mode == null) {
+                        Util.errorWithFileLocation(
+                            callExpression,
+                            log,
+                            "Failed to find the corresponding mode. Ignoring the entire `os.chmod` call..",
+                        )
+                        return
+                    }
+                    newFileSetMask(underlyingNode = callExpression, file = file, mask = mode)
+                }
+            }
+        }
+    }
+
+    /**
+     * Looks for the requested file in the [fileCache]. If none is found, a new [File] is created
+     * and added to the cache.
+     *
+     * @param fileName The name/path of the file.
+     * @param callExpression The [CallExpression] triggering the call lookup. It is used as a basis
+     *   ([File.underlyingNode]) if a new file has to be created.
+     * @return The [File] found in the cache or the new file in case it had to be created.
+     */
+    internal fun getOrCreateFile(fileName: String, callExpression: CallExpression): File {
+        return fileCache[fileName]
+            ?: newFile(underlyingNode = callExpression, fileName = fileName).also { file ->
+                fileCache += fileName to file
+            }
+    }
+
+    /**
+     * Walks the DFG backwards until a [FileOpen] node is found.
+     *
+     * Note: If multiple [File] nodes are found on different paths, one is selected a random and a
+     * warning is logged.
+     *
+     * @param expression The start node.
+     * @return The [File] node if one is found.
+     */
+    internal fun findFile(expression: Expression): File? {
+        val fileOpenPaths =
+            dataFlow(startNode = expression, direction = Backward(GraphToFollow.DFG)) {
+                it.overlays.any { overlay -> overlay is FileOpen }
```

Comment:
can't you just do `it is FileOpen` here? Is the overlay node not connected to the DFG?